panther_manager/test/plugins/test_shutdown_single_host_node.cpp

```diff
@@ -0,0 +1,134 @@
+{
+  std::map<std::string, std::string> service = {
+    {"command", "pwd"}, {"ip", "localhost"}, {"ping_for_success", "false"},
+    {"port", "22"},     {"timeout", "5.0"},  {"username", "husarion"},
+  };
+
+  RegisterNodeWithoutParams<panther_manager::ShutdownSingleHost>("ShutdownSingleHost");
+  EXPECT_THROW({ CreateTree("WrongShutdownSingleHost", service); }, BT::RuntimeError);
+}
+
+TEST_F(TestShutdownSingleHost, GoodTouchCommand)
+{
+  std::string test_file_path = testing::TempDir() + "test_panther_manager_good_touch_command";
+  std::filesystem::remove(test_file_path);
+  EXPECT_FALSE(std::filesystem::exists(test_file_path));
+  std::map<std::string, std::string> service = {
+    {"command", "touch " + test_file_path},
+    {"ip", "localhost"},
+    {"ping_for_success", "false"},
+    {"port", "22"},
+    {"timeout", "5.0"},
+    {"username", "husarion"},
+  };
+  RegisterNodeWithoutParams<panther_manager::ShutdownSingleHost>("ShutdownSingleHost");
+  CreateTree("ShutdownSingleHost", service);
+  auto & tree = GetTree();
+
+  auto status = tree.tickWhileRunning(std::chrono::milliseconds(100));
+  EXPECT_EQ(status, BT::NodeStatus::SUCCESS);
+  EXPECT_TRUE(std::filesystem::exists(test_file_path));
+
+  std::filesystem::remove(test_file_path);
+}
+
+TEST_F(TestShutdownSingleHost, Timeout)
+{
+  std::map<std::string, std::string> service = {
+    {"command", "sleep 0.5"}, {"ip", "localhost"}, {"ping_for_success", "false"},
+    {"port", "22"},           {"timeout", "0.1"},  {"username", "husarion"},
+  };
+  RegisterNodeWithoutParams<panther_manager::ShutdownSingleHost>("ShutdownSingleHost");
+  CreateTree("ShutdownSingleHost", service);
+  auto & tree = GetTree();
+
+  auto status = tree.tickWhileRunning(std::chrono::milliseconds(100));
+  EXPECT_EQ(status, BT::NodeStatus::FAILURE);
+}
+
+TEST_F(TestShutdownSingleHost, WrongCommand)
+{
+  std::map<std::string, std::string> service = {
+    {"command", "wrong_command"},
+    {"ip", "localhost"},
+    {"ping_for_success", "false"},
+    {"port", "22"},
+    {"timeout", "0.2"},
+    {"username", "husarion"},
+  };
+  RegisterNodeWithoutParams<panther_manager::ShutdownSingleHost>("ShutdownSingleHost");
+
+  CreateTree("ShutdownSingleHost", service);
+  auto & tree = GetTree();
+
+  auto status = tree.tickWhileRunning(std::chrono::milliseconds(300));
+  EXPECT_EQ(status, BT::NodeStatus::FAILURE);
+}
+
+TEST_F(TestShutdownSingleHost, WrongUser)
+{
+  std::map<std::string, std::string> service = {
+    {"command", "echo Hello World!"},
+    {"ip", "localhost"},
+    {"ping_for_success", "false"},
+    {"port", "22"},
+    {"timeout", "5.0"},
+    {"username", "wrong_user"},
+  };
+  RegisterNodeWithoutParams<panther_manager::ShutdownSingleHost>("ShutdownSingleHost");
+
+  CreateTree("ShutdownSingleHost", service);
+  auto & tree = GetTree();
+  CreateTree("ShutdownSingleHost", service);
+
+  auto status = tree.tickWhileRunning(std::chrono::milliseconds(100));
```

Comment:
```suggestion
  std::map<std::string, std::string> service = {
    {"command", "echo Hello World!"},
    {"ip", "localhost"},
    {"ping_for_success", "false"},
    {"port", "22"},
    {"timeout", "0.2"},
    {"username", "wrong_user"},
  };
  RegisterNodeWithoutParams<panther_manager::ShutdownSingleHost>("ShutdownSingleHost");

  CreateTree("ShutdownSingleHost", service);
  auto & tree = GetTree();
  CreateTree("ShutdownSingleHost", service);

  auto status = tree.tickWhileRunning(std::chrono::milliseconds(300));
```
So it fails because of the user not timeout